src/app/dialogs/edit-poll-item-dialog/edit-poll-item-dialog.component.ts

```diff
@@ -0,0 +1,188 @@
+   */
+  ngOnInit(): void {
+    if (this.pollItem instanceof MultipleChoiceItemParticipant) {
+      this.question = this.pollItem.question;
+      this.answers = this.pollItem.answers.map((answer) => answer.selectionOption);
+      this.itemType = 2;
+    }
+    if (this.pollItem instanceof QuizItemParticipant) {
+      this.question = this.pollItem.question;
+      this.answers = this.pollItem.answers.map((answer) => answer.selectionOption);
+      this.itemType = 3;
+    }
+    if (this.pollItem instanceof OpenTextItemParticipant) {
+      this.question = this.pollItem.question;
+      this.itemType = 1;
+    }
+  }
+
+  /**
+   * User clicked on 'Next' button.
+   * It handles the validity check of the entries on the current page and throws an
+   * error message or redirects the user to the next page if the user input is valid
+   */
+  handleNext(): void {
+    if (this.step === STEP_LABELS.length) {
+      this.step++;
+      this.editPollItem(this.poll.id, this.question, this.answers);
+    } else {
+      // Validity check
+      switch (this.step) {
+        case 1:
+          // Inputs are valid, set options data of selected item type
+          this.options = OPTIONS_DATA[this.itemType - 1].filter(option => option.visibleAtCreation);
+          break;
+        case 2:
+          if (this.pollItem.question === '') {
+            this.errorMessage = 'Please enter a question.';
+            return;
+          } else if (this.pollItem instanceof MultipleChoiceItemParticipant || this.pollItem instanceof QuizItemParticipant) {
+            // Multiple choice or quiz question
+            this.answers = this.trimAnswers();
+            if (this.answers[0] === '' || this.answers[1] === '') {
+              this.errorMessage = 'Please specify at least two possible answers';
+              return;
+            } else if (this.answers.length < 2) {
+              this.errorMessage = 'Please specify at least two unique answers';
+              return;
+            }
+          }
+          break;
+      }
+      // Valid input, can go to next step
+      this.step++;
+      this.errorMessage = '';
+    }
+  }
+
+  /**
+   * User clicked on the 'Back' button.
+   * Method redirects the user to the previous wizard page.
+   */
+  handleBack(): void {
+    this.step--;
+    this.errorMessage = '';
+  }
+
+  /**
+   * User clicked on the 'Cancel' button.
+   * Method closes the dialog to cancel the creation operation.
+   */
+  handleCancel(): void {
+    if (!this.loading) this.finish.emit(false);
+  }
+
+  /**
+   * Trims the list of entered answers to get rid of empty answers and duplicates.
+   * Note: The list has to be at least two elements long to display the correct UI.
+   */
+  trimAnswers(): string[] {
+    const trimmed = this.answers
+      .map(v => v.trim()) // Remove blanks or tabs from the end of each answer
+      .filter((v, i, a) => a.indexOf(v) === i) // Filter out dupes
+      .filter(v => v !== ''); // Filter out blank items
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/livepoll/live-poll/pullRequest?prid=7591734)
```suggestion
      .filter((v) => v !== ''); // Filter out blank items
```